alerting/src/main/kotlin/org/opensearch/alerting/aggregation/bucketselectorext/BucketSelectorIndices.kt

```diff
@@ -0,0 +1,79 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ *
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.alerting.aggregation.bucketselectorext
+
+import org.opensearch.common.io.stream.StreamOutput
+import org.opensearch.common.xcontent.ToXContent.Params
+import org.opensearch.common.xcontent.XContentBuilder
+import org.opensearch.search.aggregations.InternalAggregation
+import java.io.IOException
+import java.util.Objects
+
+open class BucketSelectorIndices(
+    name: String?,
+    private var parentBucketPath: String,
```

Comment:
Just for my learning: Why are we having `parentBucketPath` and `bucketIndices` one as a private variable and other as a public variable?

Reply:
`Name` and `metadata` are being inherited from the parent class `InternalAggregation` here and are `protected`. `parentBucketPath` was introduced for `BucketSelectorIndices` specifically and was only used in there so we made it `private`.